### DISCUSSION TITLE
No Display, but working audio and USB midi.

### DISCUSSION THREAD

**Discussion by randyandrade** [probonopd/MiniDexed#146] [Q&A] [answered]
Hi there,

I'm having an issue where I've used the default settings and I am, in fact, able to hear the synth on the audio output when I play keys, however I have absolutely nothing on my LCD display. I've got my breadboard layed out correctly, I believe, I have tested with a multi-meter, and I am getting continuity between the LCD module and the GPIO breakout board terminals, but I still have just a blank LCD display.

I'm using a 3B+ and I have a I2S audio hat installed, but not configured (using straight default configuration for troubleshooting purposes). My controller keyboard is a Korg Taktile Triton 49.

I've attached two images, the first is a picture of my breadboard setup with a blank display, and the second is what I get on an HDMI connected display. I don't actually expect anyone to try to sort out the connections from that image, more to show that it is connected and receiving power, but the display is completely blank, even after turning or clicking / double clicking the encoder.

Anyone have any ideas what could be my issue?

Thanks,

Randy Andrade

![152FE760-CA86-48B9-98E6-B27F415D1192_1_105_c](https://user-images.githubusercontent.com/23284626/164802940-e3484a94-bc06-46ed-ae5a-a5c279f91a8e.jpeg)
![E87467DB-CA17-429A-A023-56AB7F049CF6_1_105_c](https://user-images.githubusercontent.com/23284626/164802947-7045c2e8-d581-47fe-ae3b-57f77b7596bf.jpeg)

**Comment by probonopd** [comment, marked as answer]
Hello @randyandrade. Maybe the contrast on your display is too low? Please see https://github.com/probonopd/MiniDexed/wiki/Hardware#display.

**Reply by randyandrade** [reply]
I thought I had it connected, but there was no contact.. I pushed the 10k pot into the breadboard and now it's displaying correctly! :) Thanks so much!!

![IMG_1684 (1)](https://user-images.githubusercontent.com/23284626/164815047-247d4591-416d-4abe-8680-caf37f75614a.png)

**Reply by JoaquinSainz** [reply]
Hey nice build! What is the 10k pot for?

**Reply by craigyjp** [reply]
Probably contrast

**Reply by probonopd** [reply]
Yes.

**Comment by randyandrade** [comment]
The 10k pot is put between ground and the LCD v0 for display contrast.

I’m about to crash for tonight, but tomorrow I’ll try to make another post with how things are progressing. 

My current layout is very similar to this last picture, but I’ve gotten the 10k pots off the breadboard, soldered onto 2ft, 3-pin dupont jumpers so I can attach them to a lid which will be removable for servicing (I now have a second 10k pot which sits between +5v and the LCD “A” pin for backlight brightness control). I also have a cleaner encoder connection with a 2ft 5-pin dupont jumper for the same reason. 

Maybe 2ft of “service loop” is too much, but I’d rather be able to move the lid around more than I need instead of risking it being able to move less than I need.

The case arrives Sunday. :)

**Comment by craigyjp** [comment]
Replace it with an OLED display and remove the need for contrast or
backlight. And reduce power consumption.

**Comment by randyandrade** [comment]
Do you know of a particular model which is pinout compatible with the 1602 LCD?

Amazon link if you have one?

Thanks for the suggestion, though!

I’m always interested in lowering power consumption!

**Reply by craigyjp** [reply]
I'm using a Midas OLED MCOB21605C1V-EWP it's white on black, just check the data sheet as usually pins 1 & 2 are reversed on some OLED displays. But basically any Winstar/Midas 1602 character OLED should work. It must be powered from the 3.3v line. I bought these from Farnell about 4 years ago. Lately I have been using Surrano displays on AliExpress and buydisplay.com are very good too.

![20220422_204118.jpg](https://user-images.githubusercontent.com/51026305/164883108-b3da3f83-f4f1-4454-8383-04981c6a47d6.jpg)

**Reply by probonopd** [reply]
If you are just interested in the power consumption aspect of it, you can leave away the power line for the background LED on the 16x2 display. Then you will have authentic DX7 feeling ;-)

An OLED display is of course much nicer but usually ~10x the price.

**Comment by craigyjp** [comment]
Well anything that removes stress from a power supply is good right?

**Reply by probonopd** [reply]
Depends on your power supply :) but I doubt one LED makes a difference.

**Comment by randyandrade** [comment]
That’s pretty slick looking. I thought about doing a full width rackmount like that, but it seemed a bit overkill for my tastes, so I’m going with a 10 inch x 7 inch x 3.5 inch black project box. 

I’m trying not to be too spendy, but I do like the look of that OLED module.

@probonopd What kind of power supply do you suggest? I was thinking of getting one that I saw that was designed for the pi and was a 5v @ 3A with a power switch inline. Is it safe to go much higher than that? I’m worried that a couple of my ideas might not work without more power. 

I have thoughts to use a USB-powered headphone amp to provide a headphones output, or possibly a step-up USB cable to power an active unbalanced —> balanced channel converter, to give myself xlr outputs to feed into my mixer on the other side of the room through the cable snake.

I also have a USB midi interface to provide a set of real physical In / Thru / Out DIN-5 ports. So with all of that combined, I’m worried I’ll draw too much power through the Pi and damage it.

Do you have any other thoughts on power consumption? I feel like every mA I can save (on the display, for example) would be more juice available to power ancillary equipment (like the amp or signal converter).
